### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-brandshape

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1468]
## Description
This task describes the Storybook implementation for the sd-brandshape component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-brandshape can be found here: [sd-brandshape Notes](https://www.notion.so/matuzo/sd-brandshape-104f3f6e1bc080b98a6ee44ca74f4bb2).

-----------------------------------

## Comment
**Problem:** The demo code is not ideal because it describes a decorative image and uses a wrong description.
```
<sd-brandshape variant="image" shapes='["top", "middle", "bottom"]'>
  <div class="slot slot--border slot--text h-8 w-full">Default slot</div>
  <img
    slot="image"
    style="transform:translateY(-30%);"
    src="./placeholders/images/generic.jpg"
    alt="Generic"
  />
</sd-brandshape>
```
**Solution:** the alt-attribute should be empty here, which probably applies to all images in this component. Consider enforcing it.

- [ ] @MartaPintoTeixeira @coraliefeil how should we handle this? Is the image of the background a decorative element, so we can remove the alt content? This could be something for the "Do's & Don'ts".

**Problem:** The top shape breaks on small viewports or when zoomed in at least 300% at an initial viewport width of 1280px.
```
<sd-brandshape variant="primary" shapes="[&quot;top&quot;]">
  <p>This is just a test</p>
  <img slot="image" style="transform:translateY(-30%);" src="./placeholders/images/generic.jpg" alt="Generic">
</sd-brandshape>
```
![Image](https://github.com/user-attachments/assets/1934364a-8939-490f-b675-5b4db24b469b)
**Solution:** Avoid overflowing content using CSS.

- [ ] We will remove the slot

## Notes
- off topic: the top shape-only doesn’t work with the border-primary variant.
```
<sd-brandshape variant="border-primary" shapes="[&quot;top&quot;]">
  <p>This is just a test</p>
  <img slot="image" style="transform:translateY(-30%);" src="./placeholders/images/generic.jpg" alt="Generic">
</sd-brandshape>
```
- [ ] add this information to shape docs
 
<details>
<summary><strong>Only for design</strong></summary>
<h3>Best Practices</h3>
- Avoid putting text (without a solid background color) over image variants.
- Only describe images if they provide additional beneficial information. Avoid describing presentational images.

## Related WCAG criteria
https://www.w3.org/WAI/WCAG22/quickref/#images-of-text-no-exception

### Open Questions
- [ ] Question1
- [ ] Question2
</details>

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
